hax/treemakers/posrec.py

```diff
@@ -0,0 +1,200 @@
+                                         position and bottom array hits
+
+       - s1_area_fraction_top_probability_hax: S1 AFT computed with corrected position
+
+       - x_observed_nn_tf: TensorFlow NN reconstructed x position
+       - y_observed_nn_tf: TensorFlow NN reconstructed y position
+       
+       - s1_area_upper_injection_fraction: s1 area fraction near Rn220 injection points (near PMT 131)
+       - s1_area_lower_injection_fraction: s1 area fraction near Rn220 injection points (near PMT 243)
+    """
+    __version__ = '0.15'
+    extra_branches = ['peaks.area_per_channel[260]',
+                      'peaks.hits_per_channel[260]',
+                      'peaks.n_saturated_per_channel[260]',
+                      'interactions.x', 'interactions.y', 'interactions.z']
+
+    def __init__(self):
+
+        hax.minitrees.TreeMaker.__init__(self)
+        
+        # We need to pull some stuff from the pax config
+        self.pax_config = load_configuration("XENON1T")
+        self.tpc_channels = list(range(0, 247 + 1))
+        self.confused_s1_channels = []
+        self.statistic = self.pax_config['BuildInteractions.BasicInteractionProperties'] \
+                         ['s1_pattern_statistic']
+        qes = np.array(self.pax_config['DEFAULT']['quantum_efficiencies'])
+
+        self.pattern_fitter = PatternFitter(
+            filename=utils.data_file_name(
+                self.pax_config['WaveformSimulator']['s1_patterns_file']),
+            zoom_factor=self.pax_config['WaveformSimulator'].get(
+                's1_patterns_zoom_factor', 1),
+            adjust_to_qe=qes[self.tpc_channels],
+            default_errors=(self.pax_config['DEFAULT']['relative_qe_error'] +
+                            self.pax_config['DEFAULT']['relative_gain_error'])
+        )
+
+        # For s1_aft variable
+        aftmap_filename = utils.data_file_name(hax.config["s1_aft_filename"])
+        self.aft_map = InterpolatingMap(aftmap_filename)
+        self.low_pe_threshold = 10
+
+        # load trained NN models
+        nn_model_json = utils.data_file_name(hax.config["neural_network_model"])
+        json_file_nn = open(nn_model_json, 'r')
+        loaded_model_json = json_file_nn.read()
+        json_file_nn.close()
+        loaded_nn_model = model_from_json(loaded_model_json)
+        weights_file = utils.data_file_name(hax.config['neural_network_weights'])
+        loaded_nn_model.load_weights(weights_file)
+        loaded_nn_model.compile(loss='mean_squared_error', optimizer='adam')
+        self.nn_tensorflow = loaded_nn_model
+        self.list_bad_pmts = [1, 2, 12, 26, 34, 62, 65, 79, 86, 88, 102, 118,
+                              130, 134, 135, 139, 148, 150, 152, 162, 178, 183,
+                              190, 198, 206, 213, 214, 234, 239, 244, 27, 73,
+                              91, 137, 167, 203]
+        self.ntop_pmts = 127  # How to get this automatically?
```

Comment:
https://github.com/XENON1T/pax/blob/master/pax/config/XENON1T.ini#L450
`
len(pax_config['channels_top'])
`

Reply:
Fixed in https://github.com/XENON1T/hax/pull/174/commits/d61afe00aa7246043a4fd1b58e7bbf1cef93ce48